### PR TITLE
use cds.build.register API @sap/cds-dk >= 7.5.0

### PR DESCRIPTION
**Pull request by LotharBender**

### CODE REVIEW

**Comment by LotharBender** [comment]
> What is the status of `this.baseModel` in build plugin? 😃

I've already prepared a commit :)

File: src/build.js [hunk 1]
```diff
@@ -5,16 +5,16 @@ const { path } = cds.utils;
 
 const DEFAULT_MAIN_FOLDER = "_main";
 
-module.exports = class COV2APBuildPlugin extends cds.build.BuildPlugin {
+module.exports = class COV2APBuildPlugin extends cds.build?.Plugin {
+  static taskDefaults = { src: "srv" }
```
Comment (oklemenz2) [inline src/build.js:9]: What is this good for? It is not references anywhere...

File: cds-plugin.js [hunk 2]
```diff
@@ -12,10 +12,7 @@ if (cds.env.cov2ap.plugin) {
   cds.on("bootstrap", async (app) => {
     app.use(cov2ap.singleton());
   });
-  if (cds.env.cov2ap.build && cds.build && cds.build.register) {
-    cds.build.register("cov2ap", {
```
Comment (oklemenz2) [inline cds-plugin.js:16]: I already us register, what's the difference?

File: src/build.js (hunk 1 above)
Comment (oklemenz2) [reply src/build.js:9]: Yes `taskDefaults`, but it's not used anywhere... Can I delete it?

File: cds-plugin.js (hunk 2 above)
Comment (LotharBender) [reply cds-plugin.js:16]: The old `register` version using the `impl` property is deprecated and no longer supported. I planned to delete it with cds-dk 8

Comment (oklemenz2) [reply cds-plugin.js:16]: OK. Got it. I tested it and it works for me...

**Comment by oklemenz2** [comment]
Part of commit https://github.com/cap-js-community/odata-v2-adapter/commit/dfe863789820fe4f17b0867b8a4d37166d289fbc

**Pull request closed without merging**